apps/teamsjs-diagnostic-app/src/apis/DialogCardApi.tsx

```diff
@@ -1,8 +1,116 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { ApiComponent } from '../components/sample/ApiComponents';
+import { AdaptiveCardDialogInfo, dialog } from '@microsoft/teams-js';
+import { useDragAndDrop } from '../utils/UseDragAndDrop';
+
+export const dialogCard_CheckDialogAdaptiveCardCapability = async (): Promise<void> => {
+  console.log('Executing checkDialogAdaptiveCardCapability...');
+  try {
+    const isSupported = dialog.adaptiveCard.isSupported();
+    console.log('Dialog Adaptive Card support check result:', isSupported);
+    if (isSupported) {
+      console.log('Dialog Adaptive Card module is supported. Dialog Adaptive Card is supported on all platforms except M365 Mobile and Outlook Mobile.');
+    } else {
+      console.log('Dialog Adaptive Card module is not supported. Dialog Adaptive Card is not supported on M365 Mobile or Outlook Mobile.');
+      throw new Error('Dialog Adaptive Card module is not supported');
+    }
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : 'Unknown error';
+    console.log('Error checking Dialog Adaptive Card capability:', errorMessage);
+    if (error instanceof Error) {
+      console.log('Stack trace:', error.stack);
+    }
+    throw error;
+  }
+};
+
+export const dialogCard_OpenAdaptiveCardDialog = (input: AdaptiveCardDialogInfo): Promise<void> => {
+  console.log('Executing openAdaptiveCardDialog with input:', input);
+  return new Promise((resolve, reject) => {
+    try {
+      console.log('Opening adaptive card dialog...');
+      const onComplete = (resultObj: dialog.ISdkResponse): void => {
+        if (resultObj.err) {
+          if (resultObj.err === 'User cancelled/closed the task module.') {
```

Comment:
In user facing strings from the dialog api we should refer to them as "dialogs" and not "task modules", which is the older Teams terminology (that still exists in places in the code)

Reply:
This still says task module instead of dialog, was that intentional?

Reply:
Oops no I didn't see I had task module written in 2 places. I'll fix that!

Reply:
@ava-vellozzi pointed out to me that this message has to say 'task module' because it's in the error message that is thrown from the function